src/systemctl/systemctl.c

```diff
@@ -2428,12 +2428,25 @@ static int unit_file_find_path(LookupPaths *lp, const char *unit_name, char **un
 
         STRV_FOREACH(p, lp->search_path) {
                 _cleanup_free_ char *path;
+                _cleanup_free_ char *lpath = NULL;
+                int r;
 
                 path = path_join(arg_root, *p, unit_name);
                 if (!path)
                         return log_oom();
 
-                if (access(path, F_OK) == 0) {
+                if (access(path, F_OK) != 0)
+                        continue;
+
+                r = readlink_and_make_absolute(path, &lpath);
```

Comment:
oh, and chase_symlinks() internally acts as access(F_OK) too, hence this can be dropped then